### PR TITLE
Cleanup producer and consumer stopped children

### PR DESCRIPTION
**Pull request by jeffgrunewald**
Submitted for your approval: we have a need for dynamically managing a large number of topics and thus producers and consumers that may or may not require being restarted. Even when topic producers and consumers are not expected to be restarted, this should prevent a significant number of worker process entries from quickly expanding the size of the managing ETS table.

This should have no impact on workers if they are not stopped by the associated _stop_ function.

### CODE REVIEW

File: src/brod_producer.erl [hunk 1]
```diff
@@ -383,6 +384,9 @@ handle_cast(_Cast, #state{} = State) ->
 code_change(_OldVsn, #state{} = State, _Extra) ->
   {ok, State}.
 
+terminate(shutdown, #state{client_pid = ClientPid, topic = Topic, partition = Partition}) ->
+  brod_client:deregister_producer(ClientPid, Topic, Partition),
```
Comment (zmstone) [inline src/brod_producer.erl]: please check terminate reason like what's done in `brod_consumer:terminate`

**Comment by zmstone** [comment]
thanks for the PR.
please fix the `line too long` errors (see travis).

File: src/brod_producer.erl (hunk 1 above)
Comment (jeffgrunewald) [reply src/brod_producer.erl]: happy to; thanks for the feedback!

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
approved.
Please update `changelog.md` in the `3.8.0` section (will tag it later after, have some more PRs to merge)
Added `kc` team to review.

**Review by mikpe — APPROVED** [review]

**Review by mikpe — APPROVED** [review]

**Comment by zmstone** [comment]
squash merged. thank you for your contribution @jeffgrunewald

**Comment by jeffgrunewald** [comment]
> squash merged. thank you for your contribution @jeffgrunewald

My pleasure and thank you!

**Pull request merged**